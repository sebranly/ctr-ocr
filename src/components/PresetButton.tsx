import * as React from 'react';
import { CTR_MAX_PLAYERS } from '../constants';
import { Sign } from '../types';
import { createArraySameValue, isEqual } from '../utils/array';
import { getAbsolutePointsScheme } from '../utils/points';

export interface PresetButtonProps {
  name: string;
  isDisabledUI: boolean;
  pointsScheme: number[];
  nbPlayers: number;
  nbTeams: number;
  isRanked: boolean;
  isDoubleRush: boolean;
  setSignPointsScheme: React.Dispatch<React.SetStateAction<Sign[]>>;
  setAbsolutePointsScheme: React.Dispatch<React.SetStateAction<number[]>>;
}

const PresetButton: React.FC<PresetButtonProps> = (props) => {
  const {
    isDisabledUI,
    isDoubleRush,
    isRanked,
    nbPlayers,
    nbTeams,
    pointsScheme,
    setAbsolutePointsScheme,
    setSignPointsScheme
  } = props;
  const newAbsolutePointsScheme = getAbsolutePointsScheme(nbPlayers, nbTeams, isRanked, isDoubleRush);
  const isSamePreset = isEqual(pointsScheme.slice(0, nbPlayers), newAbsolutePointsScheme.slice(0, nbPlayers));
  return (
    <button
      onClick={() => {
        setSignPointsScheme(createArraySameValue(CTR_MAX_PLAYERS, Sign.Positive));
        setAbsolutePointsScheme(newAbsolutePointsScheme);
      }}
      disabled={isDisabledUI || isSamePreset}
    >
      FFA preset
    </button>
  );
};

export { PresetButton };
