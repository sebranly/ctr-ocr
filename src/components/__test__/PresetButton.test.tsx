import * as React from 'react';
import { render } from '@testing-library/react';

import { PresetButton } from '../PresetButton';

test('PresetButton', () => {
  const createComponent = () => (
    <PresetButton
      isDisabledUI={false}
      name="Some preset"
      pointsScheme={[8, 7, 6, 5, 4, 3, 2, 1]}
      nbPlayers={8}
      nbTeams={2}
      isRanked={false}
      isDoubleRush={false}
      setSignPointsScheme={() => {}}
      setAbsolutePointsScheme={() => {}}
    />
  );

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
