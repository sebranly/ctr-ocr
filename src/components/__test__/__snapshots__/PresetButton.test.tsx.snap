// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PresetButton 1`] = `
NodeList [
  <button
    class="ml"
  >
    FFA preset
  </button>,
]
`;
